perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/CreationEventProcessor.java

```diff
@@ -69,42 +79,39 @@ public void processEvent(String msg, MessageBeans beans) {
 	}
 
 	/**
-	 * Get entityID value from perun by facilityId.
+	 * Get facility attributes as map of LDAP attr names to their values. Key in a map is omitted if value is empty
 	 *
 	 * @param facilityId the facilityId
-	 * @return value of entityID or null, if value is null or user not exists yet
+	 * @return value of facility attributes stored in resource or null, if value is null or facility not exists yet
 	 * @throws InternalErrorException if some exception is thrown from RPC
 	 */
-	private String getFacilityEntityIdValue(int facilityId) throws InternalErrorException {
-		Perun perun = ldapcManager.getPerunBl();
+	private Map<String,String> getFacilityAttributes(int facilityId) throws InternalErrorException {
```

Comment:
Please make this method return the list of PerunAttribute objects. This way you can use existing mechanism of mapContext/extractors to obtain the ldap attribute names and values.

Reply:
It now returns ``List<Attribute>`` which is passed to the sync operation, which was already implemented.